tartiflette/schema/schema.py

```diff
@@ -582,8 +631,217 @@ def _validate_directive_implementation(self) -> List[str]:
                 attr = getattr(directive.implementation, expected, None)
                 if attr and not is_valid_coroutine(attr):
                     errors.append(
-                        f"Directive {directive.name} Method {expected} is not awaitable"
+                        f"Directive {directive.name} Method "
+                        f"{expected} is not awaitable"
+                    )
+        return errors
+
+    def _validate_enum_extensions(self) -> List[str]:
+        errors = []
+
+        for extension in [
+            x
+            for x in self.extensions
+            if isinstance(x, GraphQLEnumTypeExtension)
+        ]:
+
+            extended = self.type_definitions.get(extension.name)
+            ext_errors = _validate_extension(
+                extended, extension.name, GraphQLEnumType, "ENUM"
+            )
+            errors.extend(ext_errors)
+            if not ext_errors:
+                values = [x.name for x in extended.values]
+                for value in extension.values:
+                    if value.name in values:
+                        errors.append(
+                            f"Can't add < {value.name} > Value "
+                            f"to < {extension.name} > "
+                            f"ENUM, cause value already exists."
+                        )
+
+                errors.extend(
+                    _validate_extension_directives(extension, extended, "ENUM")
+                )
+
+        return errors
+
+    def _validate_object_extensions(self) -> List[str]:
+        errors = []
+
+        for extension in [
+            x
+            for x in self.extensions
+            if isinstance(x, GraphQLObjectTypeExtension)
+        ]:
+
+            extended = self.type_definitions.get(extension.name)
+            ext_errors = _validate_extension(
+                extended, extension.name, GraphQLObjectType, "TYPE"
+            )
+            errors.extend(ext_errors)
+            if not ext_errors:
+                for field in extension.fields:
+                    if field in extended.implemented_fields:
+                        errors.append(
+                            f"Can't add Field < {field} > to "
+                            f"TYPE < {extended.name} > "
+                            f"cause field already exists."
+                        )
+
+                for interface in extension.interfaces:
```

Comment:
Using `already_defined` function, this could be rewrite as:
```python
errors.extend(
    already_defined(
        extended.interfaces_names,
        extension.interfaces,
        extended.name,
        "Can't add interface < {existing_value} > to type < {extended_name} > cause interface already implemented.",
    )
)
```